relay-server/src/processing/spans/process.rs

```diff
@@ -125,6 +146,131 @@ fn normalize_span(
+                scrub_db_query(
+                    raw_query,
+                    sub_op,
+                    db_system,
+                    db_operation,
+                    collection_name,
+                    span_origin,
+                )
+            } else {
+                (None, None)
+            };
+
+            let db_operation = if db_operation.is_none() {
+                match sub_op {
+                    "redis" => {
+                        // This only works as long as redis span descriptions contain the command + " *"
+                        if let Some(query) = normalized_db_query.as_ref() {
+                            let command = query.replace(" *", "");
+                            if command.is_empty() {
+                                None
+                            } else {
+                                Some(command)
+                            }
+                        } else {
+                            None
+                        }
+                    }
+                    _ => {
+                        if let Some(raw_query) = raw_query {
+                            // For other database operations, try to get the operation from data
+                            sql_action_from_query(raw_query).map(|a| a.to_uppercase())
+                        } else {
+                            None
+                        }
+                    }
+                }
+            } else {
+                db_operation.map(|db_operation| db_operation.to_uppercase())
+            };
+
+            let db_collection_name: Option<String> = if collection_name.is_none() {
+                if span_origin == Some("auto.db.supabase") {
+                    normalized_db_query
+                        .as_ref()
+                        .and_then(|query| query.strip_prefix("from("))
+                        .and_then(|s| s.strip_suffix(")"))
+                        .map(String::from)
+                } else if let Some(raw_query) = raw_query {
+                    sql_tables_from_query(raw_query, &parsed_sql)
+                } else {
+                    None
+                }
+            } else if let Some(collection_name) = collection_name
+                && db_system == Some("mongodb")
+            {
+                if let Cow::Owned(collection_name) =
+                    TABLE_NAME_REGEX.replace_all(collection_name, "{%s}")
+                {
+                    Some(collection_name)
+                } else {
+                    None
+                }
+            } else {
+                None
+            };
+
+            (normalized_db_query, db_operation, db_collection_name)
+        })
+        .unwrap_or_default();
+
+    if let Some(attributes) = attributes.value_mut() {
+        if let Some(normalized_db_query) = normalized_db_query {
+            attributes.insert(consts::NORMALIZED_DB_QUERY, normalized_db_query);
```

Comment:
We now always run this logic, even if there is already the necessary attribute, considering this is a costly operation, I'd suggest following what the other eap normalization functions is, to check for presence of attributes before running potentially running an expensive operation. Unless there is a reason to always run it, but then this should be documented.